app/main.py

```diff
@@ -1,8 +1,49 @@
+from typing import List, Dict, Union, Optional
+
+
 class Person:
-    # write your code here
-    pass
+    # Class-level dictionary to store all Person instances by name
+    people: Dict[str, "Person"] = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        """
+        Initialize a Person instance with a name and age.
+        Add the instance to the class-level people dictionary.
+        """
+        self.name: str = name
+        self.age: int = age
+        # Register the Person instance in the class attribute people
+        Person.people[name] = self
+
+    def __setattr__(self, key: str, value: Optional["Person"]) -> None:
+        """
+        Override __setattr__ to dynamically add wife/husband attributes
+        only if they are set outside the constructor.
+        """
+        if key in {"wife", "husband"} and value is not None:
+            super().__setattr__(key, value)
+        else:
+            super().__setattr__(key, value)
+
+
+def create_person_list(people: List[Dict[str, Union[(
+        str, int, None)]]]) -> List[Person]:
+    """
+    Convert a list of dictionaries into a list of Person instances.
+    Link 'wife' or 'husband' attributes where applicable.
+    """
+    # Step 1: Create all Person instances first
+    person_list: List[Person] = [Person(person["name"], person[(
+        "age")]) for person in people]
```

Comment:
The line break in the middle of the dictionary key access is unnecessary and makes the code harder to read. Consider keeping the dictionary key access on a single line for better readability.